### ISSUE TITLE
Can't rerun oc cluster up with --service-catalog=true

### ISSUE THREAD

**Issue by jmontleon** [openshift/origin#19139]
Can't rerun oc cluster up with --service-catalog=true

##### Version
oc version
oc v3.10.0-alpha.0+1691e4d-400
kubernetes v1.9.1+a0ce1bc657
features: Basic-Auth GSSAPI Kerberos SPNEGO

##### Steps To Reproduce
1.
```
$ oc cluster up --routing-suffix=172.18.0.1.nip.io --service-catalog=true --tag=latest
Starting OpenShift using openshift/origin:latest ...
I0328 18:15:46.883832    6704 config.go:38] Running "create-master-config"
I0328 18:15:49.167239    6704 config.go:45] Running "create-node-config"
I0328 18:15:50.641444    6704 flags.go:31] Running "create-kubelet-flags"
I0328 18:15:51.587321    6704 run_kubelet.go:48] Running "start-kubelet"
I0328 18:15:52.048908    6704 run_self_hosted.go:156] Waiting for the kube-apiserver to be ready.
I0328 18:16:14.053400    6704 apply_template.go:77] Installing "openshift-apiserver"
I0328 18:16:14.053418    6704 apply_template.go:77] Installing "kube-proxy"
I0328 18:16:14.053438    6704 apply_template.go:77] Installing "kube-dns"
I0328 18:16:16.173532    6704 interface.go:41] Finished installing "kube-proxy" "kube-dns" "openshift-apiserver"
I0328 18:16:55.216613    6704 run_self_hosted.go:196] openshift-apiserver available
I0328 18:16:55.216693    6704 apply_template.go:77] Installing "openshift-controller-manager"
I0328 18:16:56.964988    6704 interface.go:41] Finished installing "openshift-controller-manager"
I0328 18:16:57.075527    6704 apply_list.go:48] Installing "openshift/centos7"
I0328 18:16:57.075556    6704 apply_list.go:48] Installing "openshift/jenkins pipeline persistent"
I0328 18:16:57.075575    6704 apply_list.go:48] Installing "openshift/nodejs quickstart"
I0328 18:16:57.075600    6704 apply_list.go:48] Installing "openshift/rails quickstart"
I0328 18:16:57.075617    6704 apply_list.go:48] Installing "openshift/django quickstart"
I0328 18:16:57.075631    6704 apply_list.go:48] Installing "kube-system/prometheus"
I0328 18:16:57.075661    6704 apply_list.go:48] Installing "openshift-infra/web console server template"
I0328 18:16:57.075660    6704 apply_list.go:48] Installing "openshift/mysql"
I0328 18:16:57.075679    6704 apply_list.go:48] Installing "openshift-infra/template service broker registration"
I0328 18:16:57.075576    6704 apply_list.go:48] Installing "openshift/dancer quickstart"
I0328 18:16:57.075708    6704 apply_list.go:48] Installing "openshift/postgresql"
I0328 18:16:57.075709    6704 apply_list.go:48] Installing "openshift/mongodb"
I0328 18:16:57.075745    6704 apply_list.go:48] Installing "openshift/sample pipeline"
I0328 18:16:57.075749    6704 apply_list.go:48] Installing "openshift-infra/template service broker apiserver"
I0328 18:16:57.075612    6704 apply_list.go:48] Installing "openshift/cakephp quickstart"
I0328 18:16:57.075921    6704 apply_list.go:48] Installing "openshift-infra/template service broker rbac"
I0328 18:16:57.076988    6704 apply_list.go:48] Installing "openshift-infra/service catalog"
I0328 18:16:57.076990    6704 apply_list.go:48] Installing "openshift/mariadb"
I0328 18:16:57.075644    6704 apply_list.go:48] Installing "kube-system/heapster standalone"
I0328 18:16:57.090295    6704 registry_install.go:56] Running "openshift-image-registry"
scc "privileged" added to: ["system:serviceaccount:default:registry"]
I0328 18:17:03.422863    6704 interface.go:41] Finished installing "openshift/centos7" "openshift/cakephp quickstart" "openshift/dancer quickstart" "openshift/nodejs quickstart" "openshift/rails quickstart" "openshift/jenkins pipeline persistent" "openshift/mysql" "openshift/postgresql" "openshift/django quickstart" "openshift/sample pipeline" "openshift/mongodb" "openshift/mariadb" "kube-system/prometheus" "kube-system/heapster standalone" "openshift-infra/template service broker rbac" "openshift-infra/template service broker registration" "openshift-infra/service catalog" "openshift-infra/web console server template" "openshift-infra/template service broker apiserver" "openshift-image-registry"
I0328 18:17:03.425361    6704 admin.go:48] Running "install-router"
I0328 18:17:12.816327    6704 apply_template.go:77] Installing "tsb-apiserver"
I0328 18:17:22.806751    6704 apply_template.go:77] Installing "tsb-registration"
OpenShift server started.

The server is accessible via web console at:
    https://127.0.0.1:8443

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin
```

2.
```
oc cluster down
```

3.
```
$ oc cluster up --routing-suffix=172.18.0.1.nip.io --service-catalog=true --tag=latest
Starting OpenShift using openshift/origin:latest ...
I0328 18:18:06.290503   20742 flags.go:31] Running "create-kubelet-flags"
I0328 18:18:07.212581   20742 run_kubelet.go:48] Running "start-kubelet"
I0328 18:18:07.676125   20742 run_self_hosted.go:156] Waiting for the kube-apiserver to be ready.
I0328 18:18:26.680112   20742 apply_template.go:77] Installing "openshift-apiserver"
I0328 18:18:26.680120   20742 apply_template.go:77] Installing "kube-proxy"
I0328 18:18:26.680126   20742 apply_template.go:77] Installing "kube-dns"
I0328 18:18:36.351499   20742 interface.go:41] Finished installing "kube-proxy" "kube-dns" "openshift-apiserver"
I0328 18:18:36.378841   20742 run_self_hosted.go:196] openshift-apiserver available
I0328 18:18:36.378939   20742 apply_template.go:77] Installing "openshift-controller-manager"
I0328 18:18:38.112255   20742 interface.go:41] Finished installing "openshift-controller-manager"
I0328 18:18:38.171936   20742 apply_list.go:48] Installing "openshift/postgresql"
I0328 18:18:38.171982   20742 apply_list.go:48] Installing "openshift/jenkins pipeline persistent"
I0328 18:18:38.172023   20742 apply_list.go:48] Installing "kube-system/heapster standalone"
I0328 18:18:38.172054   20742 apply_list.go:48] Installing "kube-system/prometheus"
I0328 18:18:38.172034   20742 apply_list.go:48] Installing "openshift-infra/template service broker registration"1. [step 1]
2. [step 2]
I0328 18:18:38.172078   20742 apply_list.go:48] Installing "openshift-infra/service catalog"
I0328 18:18:38.171950   20742 apply_list.go:48] Installing "openshift/sample pipeline"
I0328 18:18:38.172024   20742 apply_list.go:48] Installing "openshift/mariadb"
I0328 18:18:38.171989   20742 apply_list.go:48] Installing "openshift-infra/web console server template"
I0328 18:18:38.171989   20742 apply_list.go:48] Installing "openshift/mongodb"
I0328 18:18:38.173026   20742 apply_list.go:48] Installing "openshift-infra/template service broker rbac"
I0328 18:18:38.173154   20742 apply_list.go:48] Installing "openshift/nodejs quickstart"
I0328 18:18:38.172771   20742 apply_list.go:48] Installing "openshift-infra/template service broker apiserver"
I0328 18:18:38.172749   20742 apply_list.go:48] Installing "openshift/django quickstart"
I0328 18:18:38.173301   20742 apply_list.go:48] Installing "openshift/cakephp quickstart"
I0328 18:18:38.172961   20742 apply_list.go:48] Installing "openshift/rails quickstart"
I0328 18:18:38.171956   20742 apply_list.go:48] Installing "openshift/centos7"
I0328 18:18:38.175101   20742 apply_list.go:48] Installing "openshift/dancer quickstart"
I0328 18:18:38.173090   20742 apply_list.go:48] Installing "openshift/mysql"
I0328 18:18:43.622877   20742 interface.go:41] Finished installing "openshift/centos7" "openshift/sample pipeline" "openshift/mongodb" "openshift/mariadb" "openshift/postgresql" "openshift/cakephp quickstart" "openshift/dancer quickstart" "openshift/django quickstart" "openshift/mysql" "openshift/nodejs quickstart" "openshift/rails quickstart" "openshift/jenkins pipeline persistent" "kube-system/prometheus" "kube-system/heapster standalone" "openshift-infra/template service broker rbac" "openshift-infra/template service broker registration" "openshift-infra/service catalog" "openshift-infra/web console server template" "openshift-infra/template service broker apiserver" "openshift-image-registry"
Error: FAIL
   Error: cannot instantiate service catalog template
   Caused By:
     Error: cannot create objects from template openshift-infra/service-catalog
     Caused By:
       Error: Service "apiserver" is invalid: spec.clusterIP: Invalid value: "172.30.1.2": provided IP is already allocated


##### Current Result
Failure

##### Expected Result
Cluster comes back up

##### Additional Information
To get it working again

```
$ oc cluster down
$ for i in $(mount | grep openshift | awk '{ print $3}'); do sudo umount "$i"; done && sudo rm -rf ./openshift.local.clusterup
$ oc cluster up --routing-suffix=172.18.0.1.nip.io --service-catalog=true --tag=latest
Starting OpenShift using openshift/origin:latest ...
I0328 18:20:24.176777   28081 config.go:38] Running "create-master-config"
I0328 18:20:26.968095   28081 config.go:45] Running "create-node-config"
I0328 18:20:28.226166   28081 flags.go:31] Running "create-kubelet-flags"
I0328 18:20:29.132971   28081 run_kubelet.go:48] Running "start-kubelet"
I0328 18:20:29.608225   28081 run_self_hosted.go:156] Waiting for the kube-apiserver to be ready.
I0328 18:20:51.612669   28081 apply_template.go:77] Installing "kube-dns"
I0328 18:20:51.612669   28081 apply_template.go:77] Installing "openshift-apiserver"
I0328 18:20:51.612669   28081 apply_template.go:77] Installing "kube-proxy"
I0328 18:20:53.698706   28081 interface.go:41] Finished installing "kube-proxy" "kube-dns" "openshift-apiserver"
I0328 18:21:20.740904   28081 run_self_hosted.go:196] openshift-apiserver available
I0328 18:21:20.740981   28081 apply_template.go:77] Installing "openshift-controller-manager"
I0328 18:21:22.504527   28081 interface.go:41] Finished installing "openshift-controller-manager"
I0328 18:21:22.600340   28081 apply_list.go:48] Installing "openshift/rails quickstart"
I0328 18:21:22.600366   28081 apply_list.go:48] Installing "openshift/django quickstart"
I0328 18:21:22.600361   28081 apply_list.go:48] Installing "openshift/sample pipeline"
I0328 18:21:22.600379   28081 apply_list.go:48] Installing "openshift/cakephp quickstart"
I0328 18:21:22.600403   28081 apply_list.go:48] Installing "openshift-infra/service catalog"
I0328 18:21:22.600403   28081 apply_list.go:48] Installing "openshift/nodejs quickstart"
I0328 18:21:22.600436   28081 apply_list.go:48] Installing "openshift/mariadb"
I0328 18:21:22.600444   28081 apply_list.go:48] Installing "openshift/jenkins pipeline persistent"
I0328 18:21:22.600453   28081 apply_list.go:48] Installing "openshift-infra/template service broker registration"
I0328 18:21:22.600500   28081 apply_list.go:48] Installing "kube-system/heapster standalone"
I0328 18:21:22.600503   28081 apply_list.go:48] Installing "openshift-infra/web console server template"
I0328 18:21:22.600340   28081 apply_list.go:48] Installing "openshift/centos7"
I0328 18:21:22.600409   28081 apply_list.go:48] Installing "openshift/mongodb"
I0328 18:21:22.600582   28081 apply_list.go:48] Installing "openshift-infra/template service broker rbac"
I0328 18:21:22.600620   28081 apply_list.go:48] Installing "openshift-infra/template service broker apiserver"
I0328 18:21:22.600369   28081 apply_list.go:48] Installing "openshift/dancer quickstart"
I0328 18:21:22.600477   28081 apply_list.go:48] Installing "openshift/mysql"
I0328 18:21:22.600491   28081 apply_list.go:48] Installing "kube-system/prometheus"
I0328 18:21:22.600443   28081 apply_list.go:48] Installing "openshift/postgresql"
I0328 18:21:22.611391   28081 registry_install.go:56] Running "openshift-image-registry"
scc "privileged" added to: ["system:serviceaccount:default:registry"]
I0328 18:21:28.522849   28081 interface.go:41] Finished installing "openshift/centos7" "openshift/cakephp quickstart" "openshift/django quickstart" "openshift/rails quickstart" "openshift/mongodb" "openshift/mariadb" "openshift/dancer quickstart" "openshift/nodejs quickstart" "openshift/jenkins pipeline persistent" "openshift/sample pipeline" "openshift/mysql" "openshift/postgresql" "kube-system/prometheus" "kube-system/heapster standalone" "openshift-infra/service catalog" "openshift-infra/template service broker rbac" "openshift-infra/template service broker registration" "openshift-infra/web console server template" "openshift-infra/template service broker apiserver" "openshift-image-registry"
I0328 18:21:28.559053   28081 admin.go:48] Running "install-router"
I0328 18:21:39.720021   28081 apply_template.go:77] Installing "tsb-apiserver"
I0328 18:21:47.765490   28081 apply_template.go:77] Installing "tsb-registration"
OpenShift server started.

The server is accessible via web console at:
    https://127.0.0.1:8443

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin

```

**Comment by jorgemoralespou** [comment]
oc cluster up in openshift 3.10 will no longer support that flag.

See https://github.com/openshift/origin/pull/19209

In 3.9 and previous, there is a bug, so you should not pass 2 times the --service-catalog flag if you're preserving the configuration. With the changes mentioned in that PR, this will be solved in 3.10 and later.

Hope this helps.

**Comment by jwforres** [comment]
I'm closing this given all the changes to oc cluster up
